redfish_client/resource.py

```diff
@@ -42,6 +42,23 @@ def __init__(self, connector, oid=None, data=None):
             self._content = data
             self._headers = {}
 
+        if isinstance(self._content, dict):
```

Comment:
Can data be anything else other than `None` or dict? If not, I'd go just with `if data:`